Doc/library/asyncio-sync.rst

```diff
@@ -257,12 +257,18 @@ Condition
       Once awakened, the Condition re-acquires its lock and this method
       returns ``True``.
 
+      Note that a task *may* return from this call without having been
+      explicity awoken, which is why the caller should always re-check the state
+      and be prepared to :meth:`wait` again. For this reason, you may
+      prefer to use :meth:`wait_for` instead.
```

Comment:
Yes, I am trying to emphasize that the waiter must always check the state, among other things because more tasks may be awoken than asked for in a notify() call.